airflow-core/src/airflow/migrations/versions/0042_3_0_0_add_uuid_primary_key_to_task_instance_.py

```diff
@@ -279,8 +280,9 @@ def upgrade():
         batch_op.create_unique_constraint("task_instance_composite_key", ti_fk_cols)
         batch_op.create_primary_key("task_instance_pkey", ["id"])
 
-    # Create foreign key constraints
-    create_foreign_keys()
+    # For "sqlite" we've simply deferred FK constraint evaluation, not removed the FKs.
+    if dialect_name != "sqlite":
+        create_foreign_keys()
```

Comment:
Given we have changed the PK on the `task_instance` table in this migration from `(dag_id, task_id, run_id, map_index)` to `(id)` does this work?